tests/accelerators/test_hpu.py

```diff
@@ -0,0 +1,187 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from typing import Optional
+
+import pytest
+import torch
+import torch.nn.functional as F
+
+from pytorch_lightning import Callback, seed_everything, Trainer
+from pytorch_lightning.accelerators import CPUAccelerator, HPUAccelerator
+from pytorch_lightning.core.lightning import LightningModule
+from pytorch_lightning.plugins import HPUPlugin, HPUPrecisionPlugin
+from pytorch_lightning.trainer.states import RunningStage, TrainerFn
+from pytorch_lightning.trainer.supporters import CombinedLoader
+from pytorch_lightning.utilities import _HPU_AVAILABLE, DeviceType
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from tests.helpers.boring_model import BoringModel
+from tests.helpers.datamodules import ClassifDataModule
+from tests.helpers.runif import RunIf
+from tests.helpers.simple_models import ClassificationModel
+
+
+class HPUModel(BoringModel):
+    def training_step(self, batch, batch_idx):
+        output = self(batch)
+        loss = self.loss(batch, output)
+        return loss
+
+    def validation_step(self, batch, batch_idx):
+        output = self(batch)
+        loss = self.loss(batch, output)
+        return loss
+
+    def test_step(self, batch, batch_idx):
+        output = self(batch)
+        loss = self.loss(batch, output)
+        return loss
+
+    def training_epoch_end(self, outputs) -> None:
+        pass
+
+    def validation_epoch_end(self, outputs) -> None:
+        pass
+
+    def test_epoch_end(self, outputs) -> None:
+        pass
+
+
+@pytest.mark.skipif(_HPU_AVAILABLE, reason="test requires non-HPU machine")
+def test_fail_if_no_hpus(tmpdir):
+    with pytest.raises(MisconfigurationException, match="HPU Accelerator requires HPU devices to run"):
+        Trainer(default_root_dir=tmpdir, hpus=1)
+
+    with pytest.raises(MisconfigurationException, match="HPU Accelerator requires HPU devices to run"):
+        Trainer(default_root_dir=tmpdir, hpus=1, accelerator="hpu")
+
+
+@RunIf(hpu=True)
+def test_accelerator_selected(tmpdir):
+    trainer = Trainer(default_root_dir=tmpdir, hpus=1)
+    assert isinstance(trainer.accelerator, HPUAccelerator)
+    trainer = Trainer(default_root_dir=tmpdir, hpus=1, accelerator="hpu")
+    assert isinstance(trainer.accelerator, HPUAccelerator)
+
+
+@RunIf(hpu=True)
+def test_warning_if_hpus_not_used(tmpdir):
+    with pytest.warns(UserWarning, match="HPU available but not used. Set the `hpus` flag in your trainer"):
+        Trainer(default_root_dir=tmpdir)
+
+
+## TBD
+@pytest.mark.skipif(_HPU_AVAILABLE, reason="PyTorch is not linked with support for hpu devices")
+@RunIf(hpu=True)
+@pytest.mark.parametrize("hpus", [1, 8])
+def test_all_stages(tmpdir, hpus):
+    if hpus > 1:
```

Comment:
I believe the HPU Accelerator might want to see this flag automatically.

Reply:
can be done internally too